File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/domain/product/ProductInfo.kt
package pt.isel.markettracker.domain.product

import pt.isel.markettracker.domain.City

data class ProductInfo(
    val id: String,
    val name: String,
    val imageUrl: String,
    val brand: String,
    val category: String
)

data class StoreInfo(
    val id: Int,
    val name: String,
    val address: String,
    val city: City?,
    val isOnline: Boolean,
    val company: CompanyInfo
)

data class CompanyInfo(
    val id: Int,
    val name: String,
    val logoUrl: String
)

File: src/market-tracker-webapi/Application/Repository/Dto/Product/ProductInfo.cs
using market_tracker_webapi.Application.Domain;

namespace market_tracker_webapi.Application.Repository.Dto.Product;

public record ProductInfo(
    string Id,
    string Name,
    string ImageUrl,
    int Quantity,
    string Unit,
    double Rating,
    Brand Brand,
    Category Category
)
{
    public static ProductInfo ToProductInfo(Domain.Product product, Brand brand, Category category)
    {
        return new ProductInfo(
            product.Id,
            product.Name,
            product.ImageUrl,
            product.Quantity,
            product.Unit,
            product.Rating,
            brand,
            category
        );
    }
    
    public static ProductInfo ToProductInfo(ProductDetails productDetails)
    {
        return new ProductInfo(
            productDetails.Id,
            productDetails.Name,
            productDetails.ImageUrl,
            productDetails.Quantity,
            productDetails.Unit,
            productDetails.Rating,
            productDetails.Brand,
            productDetails.Category
        );
    }
}
